rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -760,10 +777,14 @@ async fn run_controller(
                 }
             },
             resolvers = dns_listener.notified() => {
-                let resolvers = resolvers?;
-                if let Some(session) = controller.session.as_mut() {
-                    tracing::debug!(?resolvers, "New DNS resolvers, calling `Session::set_dns`");
-                    session.connlib.set_dns(resolvers).await?;
+                let mut resolvers = resolvers?;
+                resolvers.sort();
```

Comment:
I think using `HashSet::<&IpAddr>::from_iter(resolvers.iter())` to compare would express better the intent of this expression, furthermore this is losing the order of the resolvers which is not arbitrary.